### PR TITLE
added admin router 

### PR DESCRIPTION
**Pull request by yuvalelarat**
i have created admin router through this middleware:
app.use("/api", authenticateUser, validateAdmin, adminRouter);

endpoints that's been added:
1) get all users from DB
2) edit a certain user details and update in DB
3) delete a user from DB

this is what the FE team asked for now,
i have added documentation and comments let me know if its fine.
:)

### CODE REVIEW

File: src/routes/admin.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    GET / --> Default route that returns a success message with an admin role.
+    GET /users --> Retrieve all users from the database along with their roles.
+    PUT /edit/:id --> Edit a user's details (first name, last name, email, and role).
+    DELETE /delete/:id --> Delete a user from the database.
+ */
+
+const router = express.Router();
+
+// Endpoint to show a success admin entry
+router.get("/", async (req: Request, res: Response) => {
```
Comment (tomered) [inline src/routes/admin.ts]: Please define "/admin" instead of using the generic root

File: src/routes/admin.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  return res.status(200).json({
+    status: "The user's role is admin (0), success",
+  });
+});
+
+// Endpoint to get all users
+router.get("/users", async (req: Request, res: Response) => {
```
Comment (tomered) [inline src/routes/admin.ts]: Please put this route under "/admin",
it should look like this: "/admin/users"

File: src/routes/admin.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    return res.status(200).json(userResponse);
+  } catch (error) {
+    console.error("Error fetching users:", error);
+    return res.status(500).json({ message: "Server error" });
+  }
+});
+
+// Endpoint to edit a certain user
+router.put("/edit/:id", async (req: Request, res: Response) => {
```
Comment (tomered) [inline src/routes/admin.ts]: Please put this route under "/admin",
it should look like this: "/admin/edit/:id"

File: src/routes/admin.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      },
+    });
+  } catch (error) {
+    console.error("Error updating user:", error);
+    return res.status(500).json({ message: "Server error" });
+  }
+});
+
+// Endpoint to delete a certain user
+router.delete("/delete/:id", async (req: Request, res: Response) => {
```
Comment (tomered) [inline src/routes/admin.ts]: Please put this route under "/admin",
it should look like this:"/admin/delete/:id"

**Review by tomered — CHANGES_REQUESTED** [review]

**Comment by yuvalelarat** [comment]
maybe i can change this in index.ts:
app.use("/api", authenticateUser, validateAdmin, adminRouter);
to this:
app.use("/api/admin", authenticateUser, validateAdmin, adminRouter);
is this fine to you?

File: src/routes/admin.ts (hunk 4 above)
Comment (yuvalelarat) [reply src/routes/admin.ts:110]: maybe i can change this in index.ts:
app.use("/api", authenticateUser, validateAdmin, adminRouter);
to this:
app.use("/api/admin", authenticateUser, validateAdmin, adminRouter);
is this fine to you?

**Pull request merged**